example/calculate_pi/pi.hpp

```diff
@@ -49,7 +51,14 @@ using parthenon::Globals::nranks;
 
 class CalculatePi : public Driver {
  public:
-  CalculatePi(ParameterInput *pin, Mesh *pm, Outputs *pout) : Driver(pin, pm, pout) {}
+  CalculatePi(ParameterInput *pin, Mesh *pm) : Driver(pin, pm) {
+    InitializeOutputs();
+    AppInputs_t required;
+    AppInputs_t desired;
+    desired["Pi"].push_back("radius");
+    desired["graphics"].push_back("variables");
+    pin->CheckRequiredDesired(required, desired);
+  }
```

Comment:
I find it a little strange that this lives in the driver, as opposed to when the package is set in `process_packages`. I dislike that this is one extra thing that needs to be overriden per application, especially when it feels like a per-package thing instead. I think this could be done at the same time as the input file is actually read. Is there a reason not to do it that way?

Reply:
see my book below :wink: